airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -526,7 +526,7 @@ def build_pod_request_obj(self, context=None):
             api_version="v1",
             kind="Pod",
             metadata=k8s.V1ObjectMeta(
-                namespace=self.namespace,
+                namespace=self.namespace or self.hook.get_namespace(),
```

Comment:
Might need to do this at the start of execute instead. Other things (like get_or_create_pod) use the namespace too.

Reply:
yeah this is the only spot right? https://github.com/apache/airflow/blob/087a944a95998411c9752af25a3c180465ebc9ec/airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py#L384

note that i don't actually _set_ namespace....

that line... i'm not sure that using `self.namespace` there is actually necessary.  i am pretty sure `pod.metadata.namespace` will always have a value since, as we saw from slack i think, it needs one in order to be created.  so probably we could remove that reference to `self.namespace` wdyt?

Reply:
Yeah, sounds like a plan.

Reply:
ok i removed that